packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -56,14 +56,40 @@ function getProjectConfiguration(project) {
   return parsedProjectConfiguration;
 }
 
+function _addonFound(pathSegments, projectConfiguration) {
+  const addonPaths = projectConfiguration.pkg['ember-addon'].paths;
+
+  if (!addonPaths) {
+    return;
+  }
+
+  const addonNames = addonPaths.map((path) => path.replace('lib/', ''));
+
+  return pathSegments.filter((addonName) => addonNames.includes(addonName))[0];
+}
+
+function _getParsedAddonPathSegments(pathSegments, addonName) {
+  const RELATIVE_ADDON_PATH_ROOT = 1;
+
+  pathSegments[0] = 'lib';
```

Comment:
I'm a bit confused as to what's going on here. We're mutating the passed in array, and replacing it's 0th index. What is the rationale here?

Reply:
We want to change the filepath we get: `tests/fake-addon/unit/with-hooks-in-repo-addon-test.js`
into this: `lib/fake-addon/tests/unit/with-hooks-in-repo-addon-test.js`.

So we're changing the 0th pathSegment to 'lib' and inserting 'tests' right after the addon name.

But I'm looking into the node `path` API to see if it can replace all this array manipulation.

Reply:
I think this is the concern I have with using this approach (arrays for path manipulation), is that it feels unstructured and potentially error prone. None of the elements of an array have any semantic meaning, so it's really hard to review code like this. The 0th element of an array means nothing, and the fact that we have to call out the 1st index using a screaming caps constant is pretty telling here.

Reply:
I'm currently refactoring and using `path` API.

Reply:
My latest changes on Friday have refactored this to use `path` API instead of arrays.